posthog/warehouse/api/test/test_data_warehouse.py

```diff
@@ -148,3 +150,217 @@ def test_recent_activity_edge_cases(self):
         types = [r["type"] for r in data["results"]]
         self.assertIn("Stripe", types)
         self.assertIn("Materialized view", types)
+
+    def test_job_stats_default_7_days(self):
+        """Test job_stats endpoint with default 7-day period"""
+        endpoint = f"/api/projects/{self.team.id}/data_warehouse/job_stats"
+
+        source = ExternalDataSource.objects.create(
+            source_id="test-id", connection_id="conn-id", destination_id="dest-id", team=self.team, source_type="Stripe"
+        )
+        schema = ExternalDataSchema.objects.create(name="customers", team=self.team, source=source)
+
+        ExternalDataJob.objects.create(
+            pipeline_id=source.pk,
+            schema=schema,
+            team=self.team,
+            status=ExternalDataJob.Status.COMPLETED,
+            rows_synced=100,
+        )
+        ExternalDataJob.objects.create(
+            pipeline_id=source.pk,
+            schema=schema,
+            team=self.team,
+            status=ExternalDataJob.Status.FAILED,
+            rows_synced=0,
+        )
+        DataModelingJob.objects.create(team=self.team, status=DataModelingJob.Status.COMPLETED, rows_materialized=50)
+
+        response = self.client.get(endpoint)
+        data = response.json()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(data["days"], 7)
+        self.assertEqual(data["total_jobs"], 3)
+        self.assertEqual(data["successful_jobs"], 2)
+        self.assertEqual(data["failed_jobs"], 1)
+        self.assertEqual(data["external_data_jobs"]["total"], 2)
+        self.assertEqual(data["external_data_jobs"]["successful"], 1)
+        self.assertEqual(data["external_data_jobs"]["failed"], 1)
+        self.assertEqual(data["modeling_jobs"]["total"], 1)
+        self.assertEqual(data["modeling_jobs"]["successful"], 1)
+        self.assertEqual(data["modeling_jobs"]["failed"], 0)
+        self.assertIn("breakdown", data)
+        self.assertIn("cutoff_time", data)
+
+    def test_job_stats_1_day_hourly_breakdown(self):
+        """Test job_stats endpoint with 1-day period returns hourly breakdown"""
+        endpoint = f"/api/projects/{self.team.id}/data_warehouse/job_stats"
+
+        source = ExternalDataSource.objects.create(
+            source_id="test-id", connection_id="conn-id", destination_id="dest-id", team=self.team, source_type="Stripe"
+        )
+        schema = ExternalDataSchema.objects.create(name="customers", team=self.team, source=source)
+
+        job = ExternalDataJob.objects.create(
+            pipeline_id=source.pk,
+            schema=schema,
+            team=self.team,
+            status=ExternalDataJob.Status.COMPLETED,
+            rows_synced=100,
+        )
+        ExternalDataJob.objects.filter(pk=job.pk).update(created_at=timezone.now() - timedelta(hours=2))
+
+        with self.assertNumQueries(14):
```

Comment:
Offt, 14 is a fair few. I wonder if we could reduce this by using raw SQL instead of django orm. Dont need to optimize right now, but one for us to think about down the line

Reply:
I looked into the q count and as usual, it's more that there's 7 or so org/team calls from the layers of logic we have. The number of actual count calls isn't anything more than what's showing in the function